nncf/pruning/filter_pruning/algo.py

```diff
@@ -449,17 +455,28 @@ def _apply_binary_mask_to_module_weight_and_bias(module, mask, module_scope):
                 if module.bias is not None:
                     inplace_apply_filter_binary_mask(mask, module.bias, module_scope)
 
-        for minfo in self.pruned_module_groups_info.get_all_nodes():
-            _apply_binary_mask_to_module_weight_and_bias(minfo.module, minfo.operand.binary_filter_pruning_mask,
-                                                         minfo.module_scope)
-
-            # Applying mask to the BatchNorm node
-            related_modules = minfo.related_modules
-            if minfo.related_modules is not None and PrunedModuleInfo.BN_MODULE_NAME in minfo.related_modules \
-                    and related_modules[PrunedModuleInfo.BN_MODULE_NAME].module is not None:
-                bn_module = related_modules[PrunedModuleInfo.BN_MODULE_NAME].module
-                _apply_binary_mask_to_module_weight_and_bias(bn_module, minfo.operand.binary_filter_pruning_mask,
-                                                             minfo.module_scope)
+
+        # 1. Propagate masks for all modules
+        graph = self.model.get_original_graph()
+        # pylint: disable=protected-access
+        nx_graph = graph._nx_graph
+        model_pruner = ModelPruner(self.model, graph, nx_graph)
+        model_pruner.mask_propagation()
+
+        # 2. Apply masks
+        node_types = [v.op_func_name for v in NNCF_GENERAL_CONV_MODULES_DICT] + ['group_norm']
```

Comment:
Please, rename to something more clear like `masked_node_types` or `types_to_apply_mask`.